src/main/java/pls_no_shinobu/videostore/model/User.java

```diff
@@ -181,4 +181,51 @@ public boolean removeRental(String rId) {
 
         return true;
     }
+
+    public void updatePassword(String newPassword) throws IllegalArgumentException {
+        if (newPassword.equals(this.password))
+            throw new IllegalArgumentException(
+                    "New password can't be your current password, please try again");
+
+        if (newPassword.isEmpty())
+            throw new IllegalArgumentException("New password can't be empty, please try again");
+
+        this.password = newPassword;
```

Comment:
instead of doing this, this is fine
```suggestion
        setPassword(newPassword)
```

Reply:
the same with all method below